src/main/scala/io/iohk/ethereum/network/p2p/messages/CommonMessages.scala

```diff
@@ -66,77 +88,41 @@ object CommonMessages {
     def apply(
         protocolVersion: Int,
         networkId: Int,
-        totalDifficulty: BigInt,
+        chainWeight: ChainWeight,
         bestHash: ByteString,
-        genesisHash: ByteString,
-        latestCheckpointNumber: Option[BigInt] = None
-    ): Status = latestCheckpointNumber match {
-      case Some(num) =>
-        Status64(protocolVersion, networkId, totalDifficulty, num, bestHash, genesisHash)
-
-      case None =>
-        Status63(protocolVersion, networkId, totalDifficulty, bestHash, genesisHash)
-    }
-  }
-
-  sealed trait Status extends Message {
-    def protocolVersion: Int
-    def networkId: Int
-    def totalDifficulty: BigInt
-    def latestCheckpointNumber: BigInt
-    def bestHash: ByteString
-    def genesisHash: ByteString
+        genesisHash: ByteString
+    ): Status =
+      //TODO: explain, remove if possible
+      Status(Status.code63, protocolVersion, networkId, chainWeight, bestHash, genesisHash)
 
-    // Test API
-    def as63: Status63 =
-      Status63(protocolVersion, networkId, totalDifficulty, bestHash, genesisHash)
-
-    def as64: Status64 =
-      Status64(protocolVersion, networkId, totalDifficulty, latestCheckpointNumber, bestHash, genesisHash)
   }
 
-  case class Status63(
+  case class Status(
+      code: Int,
       protocolVersion: Int,
       networkId: Int,
-      totalDifficulty: BigInt,
+      chainWeight: ChainWeight,
       bestHash: ByteString,
       genesisHash: ByteString
-  ) extends Status {
-    override val code: Int = Status.code63
+  ) extends Message {
+    require(code == Status.code63 || code == Status.code64, s"Invalid code for Status: $code")
 
     override def toString: String = {
-      s"""Status63 {
+      s"""Status {
+         |code: $code
          |protocolVersion: $protocolVersion
          |networkId: $networkId
-         |totalDifficulty: $totalDifficulty
+         |chainWeight: $chainWeight
          |bestHash: ${Hex.toHexString(bestHash.toArray[Byte])}
          |genesisHash: ${Hex.toHexString(genesisHash.toArray[Byte])}
          |}""".stripMargin
     }
 
-    override val latestCheckpointNumber: BigInt = 0
-  }
+    def as63: Status =
+      copy(code = Status.code63)
 
-  case class Status64(
-      protocolVersion: Int,
-      networkId: Int,
-      totalDifficulty: BigInt,
-      latestCheckpointNumber: BigInt,
-      bestHash: ByteString,
-      genesisHash: ByteString
-  ) extends Status {
-    override val code: Int = Status.code64
-
-    override def toString: String = {
-      s"""Status64 {
-         |protocolVersion: $protocolVersion
-         |networkId: $networkId
-         |totalDifficulty: $totalDifficulty
-         |bestHash: ${Hex.toHexString(bestHash.toArray[Byte])}
-         |genesisHash: ${Hex.toHexString(genesisHash.toArray[Byte])}
-         |latestCheckpointNumber: $latestCheckpointNumber
-         |}""".stripMargin
-    }
+    def as64: Status =
```

Comment:
Yeah, I'm not super happy either. It would be easy to change as you suggest for the `Status` message, but somewhat less so for `NewBlock`, because those msgs are constructed before peer capabilities are available (ETCM-280). It's probably not a very big refactor but I'm not sure.

Another reason is to avoid changing multiple usages in tests, at least until clarify PV64 doubts, and make our minds on the final structure of those messages.

Reply:
Maybe for now we could just add a comment to re-consider this on ETCM-280 and resume the discussion there? (cc @pslaski )

Reply:
I have added relevant comments to the factories of these classes